### PR TITLE
Updates for zh-cn/zh-tw.  Korean language support.

### PR DESCRIPTION
**Pull request by hemp**
Our translation team prefers the usage of [month number]月 for month selection as this is more commonly seen (one big example is the Windows date-time picker).  This also allows better spacing for November and December which are 3 characters long and they run together.

Solution: Switch `monthsShort` for zh-cn/zh-tw to use `monthsShort` from moment.js.  Add Korean language support by porting from moment as well.  In addition a `titleFormat` key has been added for the `.datepicker-switch` header of the date picker.  This text used to be created with string concatenation and did not work well for Chinese and Korean.  Chinese and Korean provide a `titleFormat` key override and this key defaults to `"MM yyyy"` for English and other languages.

### CODE REVIEW

**Comment by hemp** [comment]
I think this is wrapped up now.  Would love feedback if there is something else to address.

**Comment by hemp** [comment]
I've received more feedback that the text created by this code, https://github.com/eternicode/bootstrap-datepicker/blob/master/js/bootstrap-datepicker.js#L912, should use a format in Chinese like `"yyyy年mm月"`.  

If I were to add a `switchFormat` key for Chinese and default it to `"MM yyyy"` for English, does that sound like a good plan?  @acrobat @eternicode

**Comment by hemp** [comment]
Updated the PR to support titleFormat.  Here are screenshots of `en` and `zh-CN`.

en
zh-cn

File: js/bootstrap-datepicker.js [hunk 1]
```diff
@@ -905,12 +905,14 @@
...
 				todaytxt = dates[this.o.language].today || dates['en'].today || '',
 				cleartxt = dates[this.o.language].clear || dates['en'].clear || '',
+                titleFormat = dates[this.o.language].titleFormat || dates['en'].titleFormat,
 				tooltip;
 			if (isNaN(year) || isNaN(month))
 				return;
 			this.picker.find('.datepicker-days thead .datepicker-switch')
-            .text(dates[this.o.language].months[month]+' '+ (this.o.maxViewMode < 2 ? '' : year));
-      this.picker.find('tfoot .today')
+                        // .text(dates[this.o.language].months[month]+' '+ (this.o.maxViewMode < 2 ? '' : year));
```
Comment (korun) [inline js/bootstrap-datepicker.js]: Why do we need this commented code in js-file?

File: js/bootstrap-datepicker.js [hunk 2]
```diff
@@ -905,12 +905,14 @@
...
+                        .text(DPGlobal.formatDate(new Date(year, month), titleFormat, this.o.language));
+            this.picker.find('tfoot .today')
```
Comment (korun) [inline js/bootstrap-datepicker.js]: I think here should be tabs. %)

File: tests/tests.html [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 <html>
     <head>
+        <meta charset="utf-8">
         <link rel="stylesheet" href="assets/qunit.css" />
         <script src="assets/qunit.js"></script>
         <!-- console.log for test failures: enable locally if you need extra debug info-->
         <!--<script src="assets/qunit-logging.js"></script>-->
         <script src="assets/jquery-1.7.1.min.js"></script>
         <script src="../js/bootstrap-datepicker.js"></script>
+        <script src="../js/locales/bootstrap-datepicker.zh-CN.js"></script>
```
Comment (korun) [inline tests/tests.html]: :stuck_out_tongue_winking_eye:

File: js/bootstrap-datepicker.js [hunk 4]
```diff
@@ -905,12 +905,13 @@
 				endMonth = this.o.endDate !== Infinity ? this.o.endDate.getUTCMonth() : Infinity,
 				todaytxt = dates[this.o.language].today || dates['en'].today || '',
 				cleartxt = dates[this.o.language].clear || dates['en'].clear || '',
+                titleFormat = dates[this.o.language].titleFormat || dates['en'].titleFormat,
```
Comment (hemp) [inline js/bootstrap-datepicker.js]: These are tabs, something is up with online diffs.

**Comment by hemp** [comment]
I think this is wrapped up unless there is more feedback. :-) @korun

**Comment by acrobat** [comment]
@hemp can you remove the tabs to space change in a few lines in `js/bootstrap-datepicker.js`? Thanks!

**Comment by hemp** [comment]
@acrobat Done!

**Comment by hemp** [comment]
Thoughts on this making 1.5.0?

**Comment by acrobat** [comment]
Ah yes! I will try to make some time today/next days to review this PR! Can you just rebase the pr against the current master?

**Comment by hemp** [comment]
Rebase done :sparkles:

File: tests/tests.html [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
     <head>
+        <meta charset="utf-8">
         <link rel="stylesheet" href="assets/qunit.css" />
         <script src="assets/qunit.js"></script>
         <!-- console.log for test failures: enable locally if you need extra debug info-->
         <!--<script src="assets/qunit-logging.js"></script>-->
         <script src="assets/jquery-1.7.1.min.js"></script>
         <script src="../js/bootstrap-datepicker.js"></script>
+        <!-- This locale is included to test usage of titleFormat when available: see tests/suites/options.js -->
+        <script src="../js/locales/bootstrap-datepicker.zh-CN.js"></script>
```
Comment (hebbet) [inline tests/tests.html]: @acrobat not sure about that line.

Comment (acrobat) [inline tests/tests.html]: what do you mean @hebbet? If the zh-Cn locale is used in the tests it needs to be included here

Comment (hemp) [inline tests/tests.html]: @hebbet yep, what @acrobat said.  The test verifies that the format code can use the default `"MM yyyy"` or a locale specific one like for Chinese `"yyyy年mm月"`

**Comment by hebbet** [comment]
ok, thanks, LGTE, merged.

**Pull request merged**